### PR TITLE
refactor layouts

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: omit
Docs: omit

Remove old legacy methods from cache, use the equivalents from package_layout.

@tags: slow

### CODE REVIEW

File: conans/client/manifest_manager.py [hunk 1]
```diff
@@ -27,8 +27,8 @@ def check_graph(self, graph, verify, interactive):
 
     def _handle_recipe(self, node, verify, interactive):
         ref = node.ref
-        export = self._cache.export(ref)
-        exports_sources_folder = self._cache.export_sources(ref)
+        export = self._cache.package_layout(ref).export()
+        exports_sources_folder = self._cache.package_layout(ref).export_sources()
```
Comment (jgsogo) [inline conans/client/manifest_manager.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/functional/command/copy_packages_test.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ class Pkg(ConanFile):
         ref = ConanFileReference.loads("pkg/0.1@lasote/stable")
         conanfile = load(client2.cache.conanfile(ref))
         self.assertNotIn("# Recipe revision 2", conanfile)
-        data = load(os.path.join(client2.cache.export_sources(ref), "myfile.txt"))
+        data = load(os.path.join(client2.cache.package_layout(ref).export_sources(), "myfile.txt"))
```
Comment (jgsogo) [inline conans/test/functional/command/copy_packages_test.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/functional/command/download_test.py [hunk 3]
```diff
@@ -36,16 +36,16 @@ class Pkg(ConanFile):
...
 
         self.assertIn("Downloading conanfile.py", client.out)
         self.assertNotIn("Downloading conan_sources.tgz", client.out)
         self.assertNotIn("Downloading conan_package.tgz", client.out)
-        export = client.cache.export(ref)
+        export = client.cache.package_layout(ref).export()
         self.assertTrue(os.path.exists(os.path.join(export, "conanfile.py")))
         self.assertEqual(conanfile, load(os.path.join(export, "conanfile.py")))
-        source = client.cache.export_sources(ref)
+        source = client.cache.package_layout(ref).export_sources()
```
Comment (jgsogo) [inline conans/test/functional/command/download_test.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/functional/command/download_test.py [hunk 4]
```diff
@@ -74,11 +74,11 @@ class Pkg(ConanFile):
 
         client.run("upload pkg/0.1@lasote/stable")
         client.run("remove pkg/0.1@lasote/stable -f")
-        self.assertFalse(os.path.exists(client.cache.export(ref)))
+        self.assertFalse(os.path.exists(client.cache.package_layout(ref).export()))
 
         client.run("download pkg/0.1@lasote/stable")
         self.assertIn("Downloading conan_sources.tgz", client.out)
-        source = client.cache.export_sources(ref)
+        source = client.cache.package_layout(ref).export_sources()
```
Comment (jgsogo) [inline conans/test/functional/command/download_test.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/functional/command/export_dirty_test.py [hunk 5]
```diff
@@ -35,7 +36,7 @@ def source(self):
         self.assertIn("pkg/1.0@user/channel: Source folder is corrupted, forcing removal",
                       client.out)
         # Check that it is empty
-        self.assertEqual(os.listdir(os.path.join(client.cache.source(ref))), [])
+        self.assertEqual(os.listdir(os.path.join(client.cache.package_layout(ref).source())), [])
```
Comment (jgsogo) [inline conans/test/functional/command/export_dirty_test.py]: `self._cache.source(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).source()`

File: conans/test/functional/command/export_dirty_test.py [hunk 6]
```diff
@@ -54,7 +55,7 @@ def setUp(self):
         self.client.run("export . lasote/stable")
         self.client.run("install Hello0/0.1@lasote/stable --build")
         ref = ConanFileReference.loads("Hello0/0.1@lasote/stable")
-        source_path = self.client.cache.source(ref)
+        source_path = self.client.cache.package_layout(ref).source()
```
Comment (jgsogo) [inline conans/test/functional/command/export_dirty_test.py]: `self._cache.source(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).source()`

File: conans/test/functional/command/export_test.py [hunk 7]
```diff
@@ -77,8 +77,8 @@ class TestConan(ConanFile):
     exports_sources = "file2.txt"
 """
         ref = ConanFileReference.loads("Hello/1.2@lasote/stable")
-        export_path = client.cache.export(ref)
-        export_src_path = client.cache.export_sources(ref)
+        export_path = client.cache.package_layout(ref).export()
+        export_src_path = client.cache.package_layout(ref).export_sources()
```
Comment (jgsogo) [inline conans/test/functional/command/export_test.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/functional/command/export_test.py [hunk 8]
```diff
@@ -182,7 +182,7 @@ class TestConan(ConanFile):
         client.current_folder = os.path.join(client.current_folder, "recipe")
         client.run("export . lasote/stable")
         ref = ConanFileReference("Hello", "1.2", "lasote", "stable")
-        export_path = client.cache.export_sources(ref)
+        export_path = client.cache.package_layout(ref).export_sources()
```
Comment (jgsogo) [inline conans/test/functional/command/export_test.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/functional/command/export_test.py [hunk 9]
```diff
@@ -225,8 +225,8 @@ class TestConan(ConanFile):
                      "file_temp.cpp": ""})
         client.run("export . lasote/stable")
         ref = ConanFileReference("Hello", "1.2", "lasote", "stable")
-        export_path = client.cache.export(ref)
-        exports_sources_path = client.cache.export_sources(ref)
+        export_path = client.cache.package_layout(ref).export()
+        exports_sources_path = client.cache.package_layout(ref).export_sources()
```
Comment (jgsogo) [inline conans/test/functional/command/export_test.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/functional/command/remove_test.py [hunk 10]
```diff
@@ -158,7 +158,7 @@ def setUp(self):
                 files["%s/%s/%s/%s" % (folder, PACKAGES_FOLDER, pack_id, CONANINFO)] = conaninfo % str(i) + "905eefe3570dd09a8453b30b9272bb44"
                 files["%s/%s/%s/%s" % (folder, PACKAGES_FOLDER, pack_id, CONAN_MANIFEST)] = ""
             files["%s/metadata.json" % folder] = fake_metadata.dumps()
-            exports_sources_dir = client.cache.export_sources(ref)
+            exports_sources_dir = client.cache.package_layout(ref).export_sources()
```
Comment (jgsogo) [inline conans/test/functional/command/remove_test.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/functional/command/search_test.py [hunk 11]
```diff
@@ -510,21 +510,21 @@ def test_cases(remote=None):
...
             shutil.copytree(origin_path, dest_path)
             server_store.update_last_revision(ref)
-            packages = client_store_path.packages(ref)
+            packages = cache.package_layout(ref).packages()
             if not os.path.exists(packages):
                 continue
             for package in os.listdir(packages):
                 pref = PackageReference(ref, package, DEFAULT_REVISION_V1)
-                origin_path = client_store_path.package(pref)
+                origin_path = cache.package_layout(ref).package(pref)
```
Comment (jgsogo) [inline conans/test/functional/command/search_test.py]: `self._cache.package(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).package()`

File: conans/test/functional/command/upload_complete_test.py [hunk 12]
```diff
@@ -95,15 +95,15 @@ def setUp(self):
                           "my_lib/debug/libd.a": "//copy",
                           "my_data/readme.txt": "//copy",
                           "my_bin/executable": "//copy"}, path=reg_folder)
-        mkdir(self.client.cache.export_sources(self.ref))
+        mkdir(self.client.cache.package_layout(self.ref).export_sources())
```
Comment (jgsogo) [inline conans/test/functional/command/upload_complete_test.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/functional/command/upload_complete_test.py [hunk 13]
```diff
@@ -95,15 +95,15 @@ def setUp(self):
...
         manifest = FileTreeManifest.create(reg_folder)
         manifest.time = '123123123'
         manifest.save(reg_folder)
         self.test_server.server_store.update_last_revision(self.ref)
 
         self.server_pack_folder = self.test_server.server_store.package(self.pref)
 
-        package_folder = self.client.cache.package(self.pref)
+        package_folder = self.client.cache.package_layout(self.ref).package(self.pref)
```
Comment (jgsogo) [inline conans/test/functional/command/upload_complete_test.py]: `self._cache.package(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).package()`

File: conans/test/functional/old/package_copier_test.py [hunk 14]
```diff
@@ -120,11 +120,11 @@ def _assert_package_doesnt_exists(self, ref, package_id, paths):
...
 
     def _create_conanfile(self, ref, paths, content="default_content"):
-        origin_reg = paths.export(ref)
+        origin_reg = paths.package_layout(ref).export()
         mkdir(origin_reg)
         save(os.path.join(origin_reg, "conanfile.py"), content)
-        save(os.path.join(paths.base_folder(ref), "metadata.json"), PackageMetadata().dumps())
-        mkdir(paths.export_sources(ref))
+        save(paths.package_layout(ref).package_metadata(), PackageMetadata().dumps())
+        mkdir(paths.package_layout(ref).export_sources())
```
Comment (jgsogo) [inline conans/test/functional/old/package_copier_test.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/functional/old/paths_test.py [hunk 15]
```diff
@@ -37,12 +37,13 @@ def basic_test(self):
...
                                      os.path.sep.join(["opencv", "2.4.10",
                                                        "lasote", "testing"]))
-        self.assertEqual(paths.base_folder(ref), expected_base)
-        self.assertEqual(paths.export(ref),
+        layout = paths.package_layout(ref)
+        self.assertEqual(layout.base_folder(), expected_base)
+        self.assertEqual(layout.export(),
                          os.path.join(expected_base, EXPORT_FOLDER))
-        self.assertEqual(paths.build(pref),
+        self.assertEqual(layout.build(pref),
```
Comment (jgsogo) [inline conans/test/functional/old/paths_test.py]: `self._cache.build(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).build()`

File: conans/test/functional/old/paths_test.py [hunk 16]
```diff
@@ -37,12 +37,13 @@ def basic_test(self):
...
                          os.path.join(expected_base, BUILD_FOLDER,  "456fa678eae68"))
-        self.assertEqual(paths.package(pref),
+        self.assertEqual(layout.package(pref),
```
Comment (jgsogo) [inline conans/test/functional/old/paths_test.py]: `self._cache.package(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).package()`

File: conans/test/functional/old/remove_old_export_sources_layout_test.py [hunk 17]
```diff
@@ -41,7 +41,7 @@ class MyPkg(ConanFile):
         self.assertEqual(os.listdir(folder), ["myfile.txt"])
         # Now install again
         client.run("install Pkg/0.1@lasote/testing --build=missing")
-        export = client.cache.export(ref)
+        export = client.cache.package_layout(ref).export()
         self.assertNotIn(EXPORT_SOURCES_DIR_OLD, os.listdir(export))
-        export_sources = client.cache.export_sources(ref)
+        export_sources = client.cache.package_layout(ref).export_sources()
```
Comment (jgsogo) [inline conans/test/functional/old/remove_old_export_sources_layout_test.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/functional/scm/scm_test.py [hunk 18]
```diff
@@ -100,7 +100,7 @@ class ConanLib(ConanFile):
         conanfile = conanfile.replace('"onesubfolder"', '"othersubfolder"')
         self.client.save({"conanfile.py": conanfile})
         self.client.run("create . user/channel")
-        folder = self.client.cache.source(self.ref)
+        folder = self.client.cache.package_layout(self.ref).source()
```
Comment (jgsogo) [inline conans/test/functional/scm/scm_test.py]: `self._cache.source(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).source()`

File: conans/test/functional/scm/scm_test.py [hunk 19]
```diff
@@ -368,7 +371,8 @@ class ConanLib(ConanFile):
         self.client.save({"conanfile.py": conanfile})
         self.client.run("create . user/channel")
 
-        folder = self.client.cache.source(ConanFileReference.loads("lib/0.1@user/channel"))
+        ref = ConanFileReference.loads("lib/0.1@user/channel")
+        folder = self.client.cache.package_layout(ref).source()
```
Comment (jgsogo) [inline conans/test/functional/scm/scm_test.py]: `self._cache.source(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).source()`

File: conans/test/functional/scm/scm_test.py [hunk 20]
```diff
@@ -399,7 +403,8 @@ class ConanLib(ConanFile):
...
```
Comment (jgsogo) [inline conans/test/functional/scm/scm_test.py]: `self._cache.source(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).source()`

File: conans/test/functional/scm/scm_test.py [hunk 21]
```diff
@@ -410,7 +415,8 @@ class ConanLib(ConanFile):
...
```
Comment (jgsogo) [inline conans/test/functional/scm/scm_test.py]: `self._cache.source(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).source()`

File: conans/test/functional/scm/scm_test.py [hunk 22]
```diff
@@ -609,7 +615,8 @@ class ConanLib(ConanFile):
         conanfile = conanfile.replace('"onesubfolder"', '"othersubfolder"')
         self.client.save({"conanfile.py": conanfile})
         self.client.run("create . user/channel")
-        folder = self.client.cache.source(ConanFileReference.loads("lib/0.1@user/channel"))
+        ref = ConanFileReference.loads("lib/0.1@user/channel")
+        folder = self.client.cache.package_layout(ref).source()
```
Comment (jgsogo) [inline conans/test/functional/scm/scm_test.py]: `self._cache.source(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).source()`

File: conans/test/functional/scm/scm_test.py [hunk 23]
```diff
@@ -663,7 +670,8 @@ def test_auto_subfolder(self):
                                                             path=self.client.current_folder))
         self.client.run("create . user/channel")
 
-        folder = self.client.cache.source(ConanFileReference.loads("lib/0.1@user/channel"))
+        ref = ConanFileReference.loads("lib/0.1@user/channel")
+        folder = self.client.cache.package_layout(ref).source()
```
Comment (jgsogo) [inline conans/test/functional/scm/scm_test.py]: `self._cache.source(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).source()`

File: conans/test/integration/export_sources_test.py [hunk 24]
```diff
@@ -78,10 +78,10 @@ def setUp(self):
...
         self.ref = ConanFileReference.loads("Hello/0.1@lasote/testing")
         self.pref = PackageReference(self.ref, NO_SETTINGS_PACKAGE_ID)
-        self.source_folder = self.client.cache.source(self.ref)
-        self.package_folder = self.client.cache.package(self.pref)
-        self.export_folder = self.client.cache.export(self.ref)
-        self.export_sources_folder = self.client.cache.export_sources(self.ref)
+        self.source_folder = self.client.cache.package_layout(self.ref).source()
+        self.package_folder = self.client.cache.package_layout(self.ref).package(self.pref)
+        self.export_folder = self.client.cache.package_layout(self.ref).export()
+        self.export_sources_folder = self.client.cache.package_layout(self.ref).export_sources()
```
Comment (jgsogo) [inline conans/test/integration/export_sources_test.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/integration/export_sources_test.py [hunk 25]
```diff
@@ -78,10 +78,10 @@ def setUp(self):
         self.client = client
         self.ref = ConanFileReference.loads("Hello/0.1@lasote/testing")
         self.pref = PackageReference(self.ref, NO_SETTINGS_PACKAGE_ID)
-        self.source_folder = self.client.cache.source(self.ref)
-        self.package_folder = self.client.cache.package(self.pref)
-        self.export_folder = self.client.cache.export(self.ref)
-        self.export_sources_folder = self.client.cache.export_sources(self.ref)
+        self.source_folder = self.client.cache.package_layout(self.ref).source()
```
Comment (jgsogo) [inline conans/test/integration/export_sources_test.py]: `self._cache.source(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).source()`

File: conans/test/integration/export_sources_test.py [hunk 26]
```diff
@@ -78,10 +78,10 @@ def setUp(self):
...
+        self.package_folder = self.client.cache.package_layout(self.ref).package(self.pref)
```
Comment (jgsogo) [inline conans/test/integration/export_sources_test.py]: `self._cache.package(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).package()`

File: conans/test/integration/export_sources_test.py [hunk 27]
```diff
@@ -263,8 +263,8 @@ def copy_test(self, mode):
 
         # new copied package data
         ref = ConanFileReference.loads("Hello/0.1@lasote/stable")
-        source_folder = self.client.cache.source(ref)
-        export_folder = self.client.cache.export(ref)
+        source_folder = self.client.cache.package_layout(ref).source()
```
Comment (jgsogo) [inline conans/test/integration/export_sources_test.py]: `self._cache.source(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).source()`

File: conans/test/integration/symlinks_test.py [hunk 28]
```diff
@@ -128,12 +128,12 @@ def export_and_copy_test(self):
...
-                       client.cache.build(pref),
-                       client.cache.package(pref),
-                       client.cache.export(team_ref),
-                       client.cache.package(team_pref)]:
+        for folder in [client.cache.package_layout(ref).export(),
+                       client.cache.package_layout(ref).source(),
+                       client.cache.package_layout(ref).build(pref),
+                       client.cache.package_layout(ref).package(pref),
+                       client.cache.package_layout(team_ref).export(),
+                       client.cache.package_layout(team_ref).package(team_pref)]:
```
Comment (jgsogo) [inline conans/test/integration/symlinks_test.py]: `short_paths=False` vs `short_paths=None`

File: conans/test/integration/symlinks_test.py [hunk 29]
```diff
@@ -183,7 +183,7 @@ class ConanSymlink(ConanFile):
             os.symlink(symlinked_path, symlink_path)
             client.run("export . danimtb/testing")
             ref = ConanFileReference("ConanSymlink", "3.0.0", "danimtb", "testing")
-            export_sources = client.cache.export_sources(ref)
+            export_sources = client.cache.package_layout(ref).export_sources()
```
Comment (jgsogo) [inline conans/test/integration/symlinks_test.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/integration/symlinks_test.py [hunk 30]
```diff
@@ -216,10 +216,10 @@ def package(self):
         os.symlink(symlinked_path, symlink_path)
         client.run("create . danimtb/testing")
         ref = ConanFileReference("ConanSymlink", "3.0.0", "danimtb", "testing")
-        cache_file = os.path.join(client.cache.export_sources(ref), "another_directory",
+        cache_file = os.path.join(client.cache.package_layout(ref).export_sources(), "another_directory",
```
Comment (jgsogo) [inline conans/test/integration/symlinks_test.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/integration/symlinks_test.py [hunk 31]
```diff
@@ -216,10 +216,10 @@ def package(self):
...
                                   "not_to_copy.txt")
         self.assertTrue(os.path.exists(cache_file))
-        cache_other_dir = os.path.join(client.cache.export_sources(ref),
+        cache_other_dir = os.path.join(client.cache.package_layout(ref).export_sources(),
```
Comment (jgsogo) [inline conans/test/integration/symlinks_test.py]: `self._cache.export_sources(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).export_sources()`

File: conans/test/unittests/client/cache_test.py [hunk 32]
```diff
@@ -49,8 +49,8 @@ def test_package_exists(self):
         layout = self.cache.package_layout(self.ref)
         self.assertFalse(layout.package_exists(pref))
 
-        mkdir(self.cache.export(self.ref))
-        mkdir(self.cache.package(pref))
+        mkdir(layout.export())
+        mkdir(layout.package(pref))
```
Comment (jgsogo) [inline conans/test/unittests/client/cache_test.py]: `self._cache.package(ref, short_paths=False)`
vs
`self._cache.package_layout(ref, short_paths=None).package()`

File: conans/client/manifest_manager.py (hunk 1 above)
Comment (memsharded) [reply conans/client/manifest_manager.py]: Actually this was a bug! It should be short_paths=None, in order to have the manifest check feature to work properly when there are short_paths.

There were other hidden bug in manifest manager (this is a not very used feature), I have also solved it, adding a test.

**Comment by memsharded** [comment]
I have reviewed the PR and your comment.

Actually it was revealing a bug! The new PackageCacheLayout default behavior of ``short_paths=None`` default is better, it will cover some cases of short_paths that are under-tested.

So I prefer to keep all the test changing the default, the idea is that if those tests suddenly were using short_paths=True, they should pass too, instead of failing. Keeping them that way also might help in future changes to the ``PackageCacheLayout`` interface, for example, if short_paths is not really passed in the constructor, but after reading the conanfile later in the flow.

**Comment by memsharded** [comment]
Extended the refactor a little bit to remove some other methods from the cache and use the package_layout ones. Please review last commit too.

File: conans/client/cache/cache.py [hunk 33]
```diff
@@ -93,19 +93,8 @@ def base_folder(self, ref):
...
-        return self.package_layout(ref).builds()
-
-    def build(self, pref, short_paths=False):
-        return self.package_layout(pref.ref, short_paths).build(pref)
-
-    def packages(self, ref):
-        return self.package_layout(ref).packages()
-
     def package(self, pref, short_paths=False):
+        # TODO: This is deprecated, only used in testing
```
Comment (jgsogo) [inline conans/client/cache/cache.py]: Add a `@deprecation.deprecated(deprecated_in="1.15", removed_in="xxx", details="...")`, or open an issue, assign it to me and I will deal with it for the future release... 😊

**Comment by jgsogo** [comment]
> for example, if short_paths is not really passed in the constructor, but after reading the conanfile later in the flow

😁😁😁

**Pull request merged**